modules/motions/ui/MotionFormStartNew/Parts/StartNewSDVTNodeOperatorRewardAddressesSet.tsx

```diff
@@ -0,0 +1,239 @@
+import { SelectControl } from 'modules/shared/ui/Controls/Select'
+import { InputControl } from 'modules/shared/ui/Controls/Input'
+import { STETH } from 'modules/blockChain/contractAddresses'
+
+type NodeOperator = {
+  id: string
+  newRewardAddress: string
+}
+
+export const formParts = createMotionFormPart({
+  motionType: MotionType.SDVTNodeOperatorRewardAddressesSet,
+  populateTx: async ({ evmScriptFactory, formData, contract }) => {
+    const sortedNodeOperators = formData.nodeOperators.sort(
+      (a, b) => Number(a.id) - Number(b.id),
+    )
+
+    const encodedCallData = new utils.AbiCoder().encode(
+      ['tuple(uint256 nodeOperatorId, address rewardAddress)[]'],
+      [
+        sortedNodeOperators.map(nodeOperator => ({
+          nodeOperatorId: Number(nodeOperator.id),
+          rewardAddress: utils.getAddress(nodeOperator.newRewardAddress),
+        })),
+      ],
+    )
+    const gasLimit = await estimateGasFallback(
+      contract.estimateGas.createMotion(evmScriptFactory, encodedCallData),
+    )
+    const tx = await contract.populateTransaction.createMotion(
+      evmScriptFactory,
+      encodedCallData,
+      { gasLimit },
+    )
+    return tx
+  },
+  getDefaultFormData: () => ({
+    nodeOperators: [
+      {
+        id: '',
+        newRewardAddress: '',
+      },
+    ] as NodeOperator[],
+  }),
+  Component: function StartNewMotionMotionFormLego({
+    fieldNames,
+    submitAction,
+  }) {
+    const { walletAddress, chainId } = useWeb3()
+    const {
+      data: nodeOperatorsList,
+      initialLoading: isNodeOperatorsDataLoading,
+    } = useSDVTNodeOperatorsList()
+
+    const trustedCaller = ContractSDVTNodeOperatorRewardAddressesSet.useSwrWeb3(
+      'trustedCaller',
+      [],
+    )
+
+    const fieldsArr = useFieldArray({ name: fieldNames.nodeOperators })
+    const { watch } = useFormContext()
+    const { isValid } = useFormState()
+    const selectedNodeOperators: NodeOperator[] = watch(
+      fieldNames.nodeOperators,
+    )
+
+    const nodeOperatorRewardAddressesMap = useMemo(() => {
+      const result: Record<number, string | undefined> = {}
+
+      if (nodeOperatorsList?.length) {
+        for (const nodeOperator of nodeOperatorsList) {
+          result[nodeOperator.id] = nodeOperator.rewardAddress
+        }
+      }
+
+      for (const nodeOperator of selectedNodeOperators) {
+        if (nodeOperator.newRewardAddress) {
+          result[parseInt(nodeOperator.id)] = utils.getAddress(
+            nodeOperator.newRewardAddress,
+          )
+        }
+      }
+
+      const invertedRecord: Record<string, number | undefined> = {}
+      for (const key in result) {
+        const value = result[key]
+        if (value) {
+          invertedRecord[value] = parseInt(key)
+        }
+      }
+
+      return invertedRecord
+    }, [nodeOperatorsList, selectedNodeOperators])
+
+    const getFilteredOptions = (fieldIdx: number) => {
+      if (!nodeOperatorsList?.length) return []
```

Comment:
Other reutrns starts form new line, maybe should add code-block for code consistently?